src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -82,9 +83,19 @@ public PdbSourceDocumentMetadataAsSourceFileProvider(IPdbFileLocatorService pdbF
                     return null;
             }
 
+            Encoding? defaultEncoding = null;
+            if (pdbCompilationOptions.TryGetValue("default-encoding", out var encodingString))
+            {
+                defaultEncoding = Encoding.GetEncoding(encodingString);
+            }
+            else if (pdbCompilationOptions.TryGetValue("fallback-encoding", out var fallbackEncodingString))
+            {
+                defaultEncoding = Encoding.GetEncoding(fallbackEncodingString);
+            }
+
             // Get text loaders for our documents. We do this here because if we can't load any of the files, then
             // we can't provide any results, so there is no point adding a project to the workspace etc.
-            var textLoaderTasks = sourceDocuments.Select(sd => _pdbSourceDocumentLoaderService.LoadSourceDocumentAsync(sd, cancellationToken)).ToArray();
+            var textLoaderTasks = sourceDocuments.Select(sd => _pdbSourceDocumentLoaderService.LoadSourceDocumentAsync(sd, defaultEncoding, cancellationToken)).ToArray();
             var textLoaders = await Task.WhenAll(textLoaderTasks).ConfigureAwait(false);
             if (textLoaders.Where(t => t is null).Any())
```

Comment:
Unrelated question: why the ".Any()" versus .All() or something? I would have assumed if one fails we'd just march on without it, hopefully?

Reply:
I was thinking more "you told me there would be 2 sources for this, but you can only supply 1, so why should I trust that one to be correct?"

Reply:
I guess I could imagine one fails due to some obscure codepage issue, but the rest can probably march on. Your call of course.

Reply:
I think for now we're better off failing here, because the fallback if this doesn't work is going to be metadata as source. If a partial type is split across two files, and one has a weird codepage issue, then returning results here would essentially block the user from viewing half of the type (ignoring the fact that this API only supports on file results _for now_). In future the editor is going to add an explicit View Metadata command, so I think this could be adjusted then. Have made a note in the tracking issue.